### ISSUE TITLE
fairing-builder failed if istio-injection=enabled on AWS EKS

### ISSUE THREAD

**Issue by aflc** [kubeflow/fairing#347]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
* Trying example notebook
  * https://github.com/kubeflow/fairing/blob/master/examples/prediction/xgboost-high-level-apis.ipynb
* `train_job.submit()` fails with a following error:

```
Using default base docker image: registry.hub.docker.com/library/python:3.6.7
Using builder: <class 'fairing.builders.cluster.cluster.ClusterBuilder'>
Building the docker image.
Building image using cluster builder.
/opt/conda/lib/python3.6/site-packages/fairing/__init__.py already exists in Fairing context, skipping...
Creating docker context: /tmp/fairing_context_jqxu315p
/opt/conda/lib/python3.6/site-packages/fairing/__init__.py already exists in Fairing context, skipping...
Not able to find aws credentials secret: aws-secret
Waiting for fairing-builder-5dpzt to start...
Waiting for fairing-builder-5dpzt to start...
Pod started running True
Not able to find aws credentials secret: aws-secret
The job fairing-job-6xnxw launched.
Error: error resolving source context: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
Usage:
  executor [flags]

Flags:
      --build-arg multi-arg type     This flag allows you to pass in ARG values at build time. Set it repeatedly for multiple values.
      --cache                        Use cache when building image
      --cache-dir string             Specify a local directory to use as a cache. (default "/cache")
      --cache-repo string            Specify a repository to use as a cache, otherwise one will be inferred from the destination provided
      --cleanup                      Clean the filesystem at the end
  -c, --context string               Path to the dockerfile build context. (default "/workspace/")
  -d, --destination multi-arg type   Registry the final image should be pushed to. Set it repeatedly for multiple destinations. (default )
  -f, --dockerfile string            Path to the dockerfile to be built. (default "Dockerfile")
      --force                        Force building outside of a container
  -h, --help                         help for executor
      --insecure                     Push to insecure registry using plain HTTP
      --insecure-pull                Pull from insecure registry using plain HTTP
      --no-push                      Do not push the image to the registry
      --reproducible                 Strip timestamps out of the image to make it reproducible
      --single-snapshot              Take a single snapshot at the end of the build.
      --skip-tls-verify              Push to insecure registry ignoring TLS verify
      --skip-tls-verify-pull         Pull from insecure registry ignoring TLS verify
      --snapshotMode string          Change the file attributes inspected during snapshotting (default "full")
      --tarPath string               Path to save the image in as a tarball instead of pushing
      --target string                Set the target build stage to build
  -v, --verbosity string             Log level (debug, info, warn, error, fatal, panic (default "info")

Waiting for fairing-job-6xnxw-xc4vq to start...
Waiting for fairing-job-6xnxw-xc4vq to start...
...
```

**What did you expect to happen:**

fairing-builder pushes our ECR repo, and fairing-job starts successfully.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

* ecr-confiig ConfigMap exists for kaniko to mount /kaniko/.docker/config.json to use [amazon-ecr-credential-helper](https://github.com/awslabs/amazon-ecr-credential-helper)
* `Not able to find aws credentials secret: aws-secret` error occurred, so manually created ECR repo `xxx.dkr.ecr.xxx.amazonaws.com/xxx/fairing-job` this time
* build context is successfully uploaded
* I don't set AWS_SECRET and using EC2 instance profile (attached to the node)

AND I found:

* **If I disabled istio-injection, it works perfectly.**
* I don't know what happened precisely, but istio-proxy may causes some network related issues

**Environment:**

- Fairing version: (use `python -c "import fairing; print(fairing.__version__)"`): 
master(c6c075de)
- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): 
v0.6.1-rc.2
- Kubernetes version: (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.7-eks-fa4c70", GitCommit:"fa4c703fa1706fac99d00d7f1f2080e9d4cc7eed", GitTreeState:"clean", BuildDate:"2019-07-23T03:14:24Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.8-eks-a977ba", GitCommit:"a977bab148535ec195f12edc8720913c7b943f9c", GitTreeState:"clean", BuildDate:"2019-07-29T20:47:04Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):
Notebook server on kubeflow (with default image: tensorflow-2.0.0a-notebook-cpu:v0.5.0)

**NOTE**: If you are using fair from master, please provide us the git commit hash.

**Comment by aflc** [comment]
I tried on kubeflow notebook server with profile (which creates a namespace with `istio-injection=enabled` label)

**Comment by anupash147** [comment]
Did not work.
